### PR TITLE
Feature/create azure function for image processing

### PR DESCRIPTION
**Pull request by talktocory**
Hi Jochen,

The world-map Azure function is ready for review.  

Some notes:

- I had to split it up into three projects because the UI and the Function both needed access to the same core services (blob, table, queue).
- It turns out the Azure Functions support DI so I was able to leverage the exact same services in both the UI and the Function.
- Azure Functions only support  up to .Net core 3.1 at this time.  Project .Net levels are as follows:
  - CampaignKit.WorldMap.Core: .Net Core 3.1
  - CampaignKit.WorldMap.UI: .Net 5.0
  - CampaignKit.WorldMap.Function: .Net Core 3.1
  - CampaignKit.WorldMap.Tests: .Net 5.0
- You will need to setup an Azure Queue named 'worldmapqueue'.
- There is no longer a .Net background process running in the UI app.  This was recreated as a DI service named MapProcessingService.
- New process works like this
  - DefaultMapRepository creates a message in 'worldmapqueue' when a new map is added to the 'worldmapmaps' table.
  - The new Azure Function is bound to the queue.  Writing to the queue kicks off the function.
  - The function calls the MapProcessingService with the mapId and the service creates the tiles.

### CODE REVIEW

**Comment by JochenLinnemann** [comment]
> * New process works like this
>   
>   * DefaultMapRepository creates a message in 'worldmapqueue' when a new map is added to the 'worldmapmaps' table.
>   * The new Azure Function is bound to the queue.  Writing to the queue kicks off the function.
>   * The function calls the MapProcessingService with the mapId and the service creates the tiles.

Is this fast enough? A function can only run for that many minutes. Shouldn't we schedule actual tiles as function calls instead of the whole tile creation process?

File: src/CampaignKit.WorldMap.Core/Services/IProgressService.cs [hunk 1]
```diff
@@ -14,16 +14,10 @@
 // limitations under the License.
 // </copyright>
 
-namespace CampaignKit.WorldMap.Services
+namespace CampaignKit.WorldMap.Core.Services
 {
-    using System;
-    using System.Linq;
     using System.Threading.Tasks;
```
Comment (JochenLinnemann) [inline src/CampaignKit.WorldMap.Core/Services/IProgressService.cs]: Can you move all usings in all files outside the namespace, please?

File: src/CampaignKit.WorldMap.Function/ProcessMapTrigger.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            this._log = log;
+            this._mapProcessingService = mapProcessingService;
+        }
+
+        /// <summary>
+        /// Runs the specified my queue item.
+        /// </summary>
+        /// <param name="myQueueItem">My queue item.</param>
+        [FunctionName("ProcessMapTrigger")]
+        public async Task Run([QueueTrigger("worldmapqueue", Connection = "")]string myQueueItem)
```
Comment (JochenLinnemann) [inline src/CampaignKit.WorldMap.Function/ProcessMapTrigger.cs:66]: The queuing service add the id as b64-encoded byte array to the queue - are we sure this automatically decoded and turned into the original UTF8 string here?

**Review by JochenLinnemann — CHANGES_REQUESTED** [review]

**Comment by talktocory** [comment]
> Is this fast enough? A function can only run for that many minutes. Shouldn't we schedule actual tiles as function calls instead of the whole tile creation process?

I had considered both approaches.  Originally I wanted to do one trigger per tile but that would mean some rewriting of the TileService.  Currently the TileService (now called DefaultMapProcessingService) creates map tiles by zoom level.  It does the following:
- Looks for unprocessed tiles
- Groups them by zoom  level
- Creates a zoom level base image
- Creates the required tiles from that zoom level base image

This doesn't quite support the one-trigger-per-tile approach.  It doesn't currently seem to be an issue because the largest image you can upload to World Map is 10MB which the function seems to handle pretty easily.  

Happy to break this piece up into multiple triggers if you want a more robust solution.  Let me know if you want me to make the changes.

**Comment by JochenLinnemann** [comment]
> > Is this fast enough? A function can only run for that many minutes. Shouldn't we schedule actual tiles as function calls instead of the whole tile creation process?
> 
> I had considered both approaches. Originally I wanted to do one trigger per tile but that would mean some rewriting of the TileService. Currently the TileService (now called DefaultMapProcessingService) creates map tiles by zoom level. It does the following:
> 
> * Looks for unprocessed tiles
> * Groups them by zoom  level
> * Creates a zoom level base image
> * Creates the required tiles from that zoom level base image
> 
> This doesn't quite support the one-trigger-per-tile approach. It doesn't currently seem to be an issue because the largest image you can upload to World Map is 10MB which the function seems to handle pretty easily.
> 
> Happy to break this piece up into multiple triggers if you want a more robust solution. Let me know if you want me to make the changes.

You are right, let's first test this before investing more work.

**Comment by talktocory** [comment]
I see that the unit tests project is now failing to build in the pipeline.  I'll look into that this weekend.

**Comment by talktocory** [comment]
@JochenLinnemann - I've fixed the Azure pipeline build.  I need to add two different SDKs as Core and Function projects use 3.1 not 5.0.

```  - task: UseDotNet@2
    displayName: Install .NET Core 5.0 SDK
    inputs:    
      packageType: sdk
      version: 5.0.x
      includePreviewVersions: true
  - task: UseDotNet@2
    displayName: Install .NET Core 3.1 SDK
    inputs:
      packageType: sdk
      version: 3.1.x
```

**Review by JochenLinnemann — APPROVED** [review]

**Pull request merged**